ultimarc/ui/devices/device.py

```diff
@@ -50,11 +48,11 @@ def __init__(self, args, env, attached,
         self.attached = attached
 
     # Implement function in child class
-    def roleNames(self):
+    def roleNames(self) -> typing.Dict:
         return None
 
     # Implement function in child class
-    def rowCount(self):
+    def rowCount(self, parent: QModelIndex | QPersistentModelIndex = ...) -> int:
         return 0
 
```

Comment:
Using the "|"  or operator type hint notation (IE: "QModelIndex | QPersistentModelIndex") only works in Python 3.10 or newer. For Python 3.9 or older an error occurs when running the code.  To allow this code to run in older versions this should be changed to "parent: Union[QModelIndex, QPersistentModelIndex]"

I see this type hint setup in each of the device code files.